magtogoek/adcp/process.py

```diff
@@ -651,6 +662,54 @@ def _load_navigation(dataset: xr.Dataset, navigation_files: str):
     return dataset
 
 
+def _regrid_dataset(dataset: xr.Dataset, pconfig: ProcessConfig) -> xr.Dataset:
+    """ Wrapper for regrid_dataset
+ 
+    Note
+    ----
+    The `interp` method performs linear interpolation. The `bin` method
+    performs averaging of input data strictly within the bin boundaries
+    and with equal weights for all data inside each bin.
+ 
+    """
+    # Pre-process flags
+    for var_ in 'uvw':
+        dataset[f"{var_}_QC"] = _prepare_flags_for_regrid(dataset[f"{var_}_QC"])
+
+    # Make new quality flags if grid_method is `bin`. Must happen before averaging.
+    if pconfig.grid_method == 'bin':
+        l.log(f"Making regridded quality flags according to bin transfer scheme.")
+
+        _bin_depths, _new_flags = np.loadtxt(pconfig.grid_depth), dict()
```

Comment:
1. Lacks flexibility ( grid_depth could be an array). 
2. is loaded again in redgrid_dataset.

Reply:
For now, grid_depth is strictly defined as a path in the config_handler. Loading the grid here is a compromise. We need it here for the bin flagging algorithm which has to happen before regridding, and the regrid_dataset will lose generality if we move the flag transfer routine inside it.

Reply:
Totally agree, for the moment no check is need in _regrid_dataset, however, _regrid_dataset could pass the loaded text data to regrid_dataset instead of the path. The pconfig.grid_depth value could be overwritten with the loaded array.

Reply:
Commit [f9a178e](https://github.com/JeromeJGuay/magtogoek/pull/5/commits/f9a178e127ee33f367e18ccc4de67f3c2a96f83e) adresses this issue. New depths are loaded once at the top of the regrid routine.